### PR TITLE
remove error from spec

### PR DESCRIPTION
**Pull request by tomahawk28**
Upgrade spec to remove errors from interface.

Slack thread: https://gigantic.slack.com/archives/C0251MXL5/p1584003641116900

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: activation_logger.go [hunk 1]
```diff
@@ -82,30 +82,26 @@ func NewActivation(config ActivationLoggerConfig) (Logger, error) {
 	return l, nil
 }
 
-func (l *activationLogger) Log(keyVals ...interface{}) error {
+func (l *activationLogger) Log(keyVals ...interface{}) {
 	activated, err := shouldActivate(l.activations, keyVals)
 	if err != nil {
-		return microerror.Mask(err)
+		panic(err)
```
Comment (tomahawk28) [inline activation_logger.go]: should we use `log` in this case or just panic?

Comment (rossf7) [reply activation_logger.go]: I think we should log. Panic will cause crashes.

Comment (fiunchinho) [reply activation_logger.go]: +1

I believe a log shouldn't panic while trying to log. Maybe we could improve the situation trying to log something with `fmt.print` to let the developer know that something is wrong.

Comment (tomahawk28) [reply activation_logger.go]: Thanks for the feedback, I'm using `log` just in case it's not working.

**Comment by tomahawk28** [comment]
perhaps @kopiczko can you look into this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
```
Comment (fiunchinho) [inline CHANGELOG.md]: It looks like unreleased would include all the versions. is there a way to make this clearer?

Comment (tomahawk28) [reply CHANGELOG.md]: It's not included any versions; https://github.com/giantswarm/micrologger/compare/v0.3.0...HEAD

Is that the same link?

Comment (fiunchinho) [reply CHANGELOG.md]: Oh nevermind

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+## [0.3.0] 2020-03-17
+
+## Changed
+
+- Remove error from the spec.
```
Comment (kopiczko) [inline CHANGELOG.md]: We talked about this in last SIG operator. This part should land in a separate PR titled "release v0.3.0". This is for the future. Thanks for sorting that out.

**Pull request merged**